src/main/java/com/limechain/trie/NodeVariant.java

```diff
@@ -32,4 +34,16 @@ public enum NodeVariant {
     public int getPartialKeyLengthHeaderMask() {
         return this.mask ^ 0xFF;
     }
+
+    // NOTE:
+    //  Technically, for NodeVariant.COMPACT_ENCODING this should return `None`, but we still return 0...
+    //  since that's the result of the computation, but it's semantically invalid
```

Comment:
Can't we add an if returning a `null` in that case?

Reply:
Sure, I didn't know this only has a singular use case.

Reply:
Well, after getting to actually refactoring this, it turned out we don't need this method at all, since what I'm using it for outside is already exactly solved by the already existing method `NodeVariant::getPartialKeyLengthHeaderMask`. You can inspect the final version here: 598e6e9, and resolve this if all is okay.

Reply:
nice